src/ipahealthcheck/dogtag/range.py

```diff
@@ -0,0 +1,170 @@
+#
+# Copyright (C) 2019 FreeIPA Contributors see COPYING for license
+#
+
+import ldap
+import logging
+
+from ipahealthcheck.dogtag.plugin import DogtagPlugin, registry
+from ipahealthcheck.core.plugin import Result
+from ipahealthcheck.core.plugin import duration
+from ipahealthcheck.core import constants
+
+from ipalib import api, errors
+from ipaplatform.paths import paths
+from ipapython import ipaldap
+from ipapython.dn import DN
+from ipapython.directivesetter import get_directive
+
+logger = logging.getLogger()
+
+
+def range_overlap(range1, range2):
+    return max(range1[0], range2[0]) <= min(range1[1], range2[1])
+
+
+@registry
+class DogtagRequestRangeCheck(DogtagPlugin):
+    """
+    Look for overlapping request ranges
+    """
+    @duration
+    def check(self):
+
+        if not self.ca.is_configured():
+            logger.debug("No CA configured, skipping dogtag config check")
+            return
+
+        try:
+            conn = ipaldap.LDAPClient.from_realm(api.env.realm)
+        except AttributeError:
+            conn = ipaldap.LDAPClient(api.env.ldap_uri)
+        conn.external_bind()
+
+        # Get the max nextRange first
+        filterstr = "(objectclass=repository)"
+        attrlist = ['nextRange', ]
+        try:
+            entries = self.conn.get_entries(
+                DN('ou=certificateRepository,ou=ca,o=ipaca'),
+                ldap.SCOPE_BASE, filterstr, attrlist)
+        except errors.EmptyResult:
+            entries = []
+        except errors.NotFound:
+            entries = []
+
+        if entries:
+            if len(entries) != 1:
+                # Can't really proceed if there are multiple so fail loudly.
+                yield Result(self, constants.CRITICAL,
+                             key='nextRange',
+                             dn='ou=certificateRepository,ou=ca,o=ipaca',
+                             msg='multiple {key} values in {dn}')
+                return
+            else:
+                nextrange = int(entries[0].single_value.get('nextRange', 0))
```

Comment:
Certificate repository (and KRA keyRepository) uses hex encoding.  Even though they may look decimal-ish :)